### DISCUSSION TITLE
Websocket connection failed

### DISCUSSION THREAD

**Discussion by saurabhgharatpara** [dymmond/esmerald#394] [Q&A]
I've added websocket handler 

@websocket(path="/ws")
async def websocket_endpoint(socket: WebSocket) -> None:
    await socket.accept()
    await socket.send_text("Hello!")
    await socket.close()
    
  and on the frontend i've called const socket = new WebSocket('ws://127.0.0.1:8000/ws')

  but im getting connection failed 403 refused
is it missing anything ?

**Comment by tarsil** [comment]
Did you declare a WebSocketGateway for that handler? Also, do you have any middlewares implemented?

**Reply by saurabhgharatpara** [reply]
Yes
WebSocketGateway(handler=websocket_endpoint, path="/ws")
and im not using any middleware

**Reply by tarsil** [reply]
I would need more details like logging. Like this is hard. When you hit the endpoint, what does it show in the console? Does it show it's trying to reach it?

**Reply by tarsil** [reply]
> Yes WebSocketGateway(handler=websocket_endpoint, path="/ws") and im not using any middleware

Its odd as we have plenty of tests for the websockets including this https://github.com/dymmond/esmerald/blob/main/tests/handlers/test_websocket_handler_gateway.py

More details are needed to understand

**Reply by saurabhgharatpara** [reply]
These are the logs
DEBUG:    = connection is CONNECTING
DEBUG:    < GET /ws HTTP/1.1
DEBUG:    < host: localhost:8000
DEBUG:    < connection: Upgrade
DEBUG:    < pragma: no-cache
DEBUG:    < cache-control: no-cache
DEBUG:    
DEBUG:    < upgrade: websocket
DEBUG:    < origin: http://127.0.0.1:8000
DEBUG:    < sec-websocket-version: 13
DEBUG:    < accept-encoding: gzip, deflate, br, zstd
DEBUG:    < accept-language: en-US,en;q=0.9
INFO:     ('127.0.0.1', 54915) - "WebSocket /ws" 403
DEBUG:    > HTTP/1.1 403 Forbidden
DEBUG:    > Date: Fri, 30 Aug 2024 04:47:10 GMT
DEBUG:    > Content-Length: 0
DEBUG:    > Content-Type: text/plain
DEBUG:    > Connection: close
DEBUG:    > [body] (0 bytes)
INFO:     connection rejected (403 Forbidden)
DEBUG:    x closing TCP connection
DEBUG:    = connection is CLOSED
INFO:     connection closed

WebSocket connection to 'ws://localhost:8000/ws' failed:
